apps/babelstream/package.py

```diff
@@ -0,0 +1,268 @@
+
+    
+    homepage = "https://github.com/UoB-HPC/BabelStream"
+    # url = "https://github.com/UoB-HPC/BabelStream/archive/refs/tags/v4.0.tar.gz"
+    # git = "https://github.com/UoB-HPC/BabelStream.git"
+    git = "https://github.com/kaanolgu/BabelStream.git" # this is temporary to have sycl2020 working before updates
+    # version("4.0", sha256="a9cd39277fb15d977d468435eb9b894f79f468233f0131509aa540ffda4f5953")
+    version('main', branch='main')
+    maintainers = ["robj0nes ", "tomdeakin", "tom91136"]
+    
+    #Languages
+    variant("sycl", default=False, description="Enable SYCL support")
+    variant("sycl2020", default=False, description="Enable SYCL support")
+    variant("omp", default=False, description="Enable OpenMP support")
+    variant("ocl", default=False, description="Enable OpenCL support")
+    variant("tbb", default=False, description="Enable TBB support")
+    variant("acc", default=False, description="Enable OpenACC support") 
+    variant("thrust", default=False, description="Enable THRUST support")
+    variant("raja", default=False, description="Enable RAJA support") 
+    variant("stddata", default=False, description="Enable STD-data support")
+    variant("stdindices", default=False, description="Enable STD-indices support")
+    variant("stdranges", default=False, description="Enable STD-ranges support")
+    
+    variant("dir", values=str, default="none", description="Enable Directory support")
+    
+    #Kokkos conflict and variant
+    conflicts("dir=none",when="+kokkos",msg="KOKKKOS requires architecture to be specfied by dir=")
+    variant("kokkos", default=False, description="Enable KOKKOS support")
+    
+    #ACC conflict
+    variant("cpu_arch", values=str, default="none", description="Enable CPU Target for ACC")
+    variant("target", values=str, default="none", description="Enable CPU Target for ACC")
+    
+    #STD conflicts
+    conflicts("+stddata",when="%gcc@:10.1.0",msg="STD-data requires newer version of GCC") 
+    conflicts("+stdindices",when="%gcc@:10.1.0",msg="STD-indices requires newer version of GCC")
+    conflicts("+stdranges",when="%gcc@:10.1.0",msg="STD-ranges requires newer version of GCC")
+    
+    # CUDA conflict 
+    conflicts("cuda_arch=none",when="+cuda",msg="CUDA requires architecture to be specfied by cuda_arch=")
+    variant("mem", values=str, default="DEFAULT", description="Enable MEM Target for CUDA")
+        
+    #download raja from https://github.com/LLNL/RAJA
+    conflicts("dir=none",when="+raja",msg="RAJA requires architecture to be specfied by dir= [#download raja from https://github.com/LLNL/RAJA]")
+
+    #Thrust Conflict
+    conflicts("~cuda", when="+thrust", msg="Thrust requires +cuda variant")
+
+    #TBB Dependency
+    depends_on("intel-oneapi-tbb", when="+tbb")
+    
+    #Kokkos Dependency
+    depends_on("kokkos@3.7.1", when="+kokkos")
+
+    #OpenCL Dependency
+    depends_on("pocl@1.5", when="+ocl")
```

Comment:
For OpenCL, we want to build against multiple OpenCL implementations:
- CUDA (provided by spack package `cuda`)
- AMD (I think provided by spack package `rocm-opencl`)
- Intel (?)
- and POCL

So this is really like a backend module, so we might want to say e.g., `+ocl +cuda`

Reply:
So something like this ? 
```
       if "+ocl" in spec:
            if("+cuda" in spec):
                cuda_dir = self.spec["cuda"].prefix
                cuda_comp = cuda_dir + "/bin/nvcc"
                args.append("-DCMAKE_CXX_COMPILER=" + cuda_comp)
                
                ````